core/arch/arm/pta/core_fs_htree_tests.c

```diff
@@ -449,17 +477,160 @@ TEE_Result core_fs_htree_tests(uint32_t nParamTypes,
 	for (n = 0; n < num_blocks; n += 3) {
 		for (m = 0; m < n; m += 3) {
 			for (o = 0; o < (n - m); o++) {
-				res = htree_test_rewrite(&aux, n, m, o);
+				res = htree_test_rewrite(aux, n, m, o);
 				CHECK_RES(res, goto out);
 				o += 2;
 			}
 		}
 	}
 
+out:
+	aux_free(aux);
+	return res;
+}
+
+static TEE_Result test_corrupt_type(const TEE_UUID *uuid, uint8_t *hash,
+				    size_t num_blocks, struct test_aux *aux,
+				    enum tee_fs_htree_type type, size_t idx)
+{
+	TEE_Result res;
+	struct test_aux aux2 = *aux;
+	struct tee_fs_htree *ht = NULL;
+	size_t offs;
+	size_t size;
+	size_t size0;
+	size_t n;
+
+	res = test_get_offs_size(type, idx, 0, &offs, &size0);
+	CHECK_RES(res, return res);
+
+	aux2.data = malloc(aux->data_alloced);
+	if (!aux2.data)
+		return TEE_ERROR_OUT_OF_MEMORY;
+
+	n = 0;
+	while (true) {
```

Comment:
Is this really testing that each single-byte corruption (at offset 0, 1, 2, 4 etc.) is detected? I don't think it does.

Reply:
Which part is it that you doubt?

Reply:
- If `do_range()` returns success for all values of `n` except the last one, the test will still pass.
- If `tee_fs_htree_open()` fails for some value of `n`, higher values won't be tested

Reply:
Thanks